src/language-html/printer-html.js

```diff
@@ -873,7 +873,9 @@ function printEmbeddedAttributeValue(node, originalTextToDoc, options) {
       const fnExpRE = /^([\w$_]+|\([^)]*?\))\s*=>|^function\s*\(/;
       const simplePathRE = /^[A-Za-z_$][\w$]*(?:\.[A-Za-z_$][\w$]*|\['[^']*?']|\["[^"]*?"]|\[\d+]|\[[A-Za-z_$][\w$]*])*$/;
 
-      const value = getValue();
+      const value = getValue()
+        // https://github.com/vuejs/vue/blob/v2.5.17/src/compiler/helpers.js#L104
+        .trim();
```

Comment:
Just curious, how is this related to the semicolon?

Reply:
We need to check if we should use `__js_expression` (no semicolon) or `babylon` (semicolon) by looking at `simplePathRE.test(value) || fnExpRE.test(value)`.